### DISCUSSION TITLE
Would it be possible to have an empty json file even when no dependencies are found ?

### DISCUSSION THREAD

**Discussion by Greybird** [tomchavakis/nuget-license#123] [Ideas]
Hi,

I noticed that when analyzing a solution where no nuget dependencies were present, and the `--json` option is specified, no json file was produced.
I can understand the reason, but in my use case where I run the tool against my whole organization code base, the non creation of the file leads to a difficulty to ensure that the solution was successfully analyzed or not.
Do you think changing the behavior of the tool to create a file with empty dependencies would be possible ? (I believe this could be interesting for all formats except console output maybe ?)

Thanks,

**Comment by tomchavakis** [comment]
Hi @Greybird,

Thank you for bringing up this idea. To be honest it's a matter of perspective. A file with empty dependencies would be possible in all formats. I guess that you are expecting a file in order to parse the response from the tool and if you are not getting a file then you are not sure if the program had successfully run the analysis or not. Can you send an example of your desired output in order to adjust the output accordingly? 

Thanks

**Comment by Greybird** [comment]
Hi @tomchavakis,

Thanks for your answer, and sorry for the very long delay to reply.

Yes, you are completely right, I'm running the tool, and even if I am trying to manage potential issues during process launch, and to handle the exit code, the presence of a file would be a far easier way to guarantee the tool has run successfully. On the contrary the absence of the file would be a perfect indicator that _something_ went wrong, which is basically the amount of details I need.

To me, a simple empty array would perfectly do as a result file, as it would be a generalization of the format used when there are dependencies.
`[]`

If the change in behavior is expected to be an issue, the option to emit a file even when no dependencies are found could be put under an option for the tool ?

Thanks,

**Comment by tomchavakis** [comment]
Hi @Greybird,

Thank you for the reply,
We can add a new option to avoid any possible issues. 
Thank you again for bringing this feature. I will create a feature in the issues and link the conversation as a reference.

**Reply by tomchavakis** [reply]
https://github.com/tomchavakis/nuget-license/issues/145
